go-runtime/ftl/singleton.go

```diff
@@ -0,0 +1,31 @@
+package ftl
+
+import (
+	"context"
+	"sync"
+)
+
+type InputLambda[T any] func(context.Context) (T, error)
+
+type SingletonConstructor[T any] struct {
+	Fn   InputLambda[T]
+	Out  T
+	Once sync.Once
```

Comment:
These should be private (lowercase).

Reply:
BTW with the Go LSP there's usually a "Rename symbol" option which will correctly rename the symbol everywhere it's used.

Reply:
Fixed